base/file.jl

```diff
@@ -424,7 +424,8 @@ function tempdir()
     temppath = Vector{UInt16}(undef, 32767)
     lentemppath = ccall(:GetTempPathW, stdcall, UInt32, (UInt32, Ptr{UInt16}), length(temppath), temppath)
     windowserror("GetTempPath", lentemppath >= length(temppath) || lentemppath == 0)
-    resize!(temppath, lentemppath)
+    isroot = ccall((:PathIsRootW,:Shlwapi), stdcall, Int32, (Ptr{UInt16},), temppath)
```

Comment:
`Shlwapi` is a very large dependency that we don't appear to use anywhere else. It would be nice avoid instantiating it just for this. There should already be a regex for this in path.jl for splitdrive.

Reply:
Sure, is there any overhead to instatitating it or is the preference to keep as much of the code in Julia? I don't see any real overhead to calling the library since it's loaded by default.

Reply:
Updated code with a julia version of `isroot`. Let me know if this is preffered